tooling/SecretManager.VisualStudio/ProjectLocalSecretsManager.cs

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Threading;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.Extensions.SecretManager.VisualStudio
+{
+    public class ProjectLocalSecretsManager : IVsProjectSecrets, SVsProjectLocalSecrets
+    {
+        private const string UserSecretsPropertyName = "UserSecretsId";
+
+        private readonly IProjectPropertiesProvider _propertiesProvider;
+
+        public ProjectLocalSecretsManager(IProjectPropertiesProvider propertiesProvider)
+        {
+            _propertiesProvider = propertiesProvider ?? throw new ArgumentNullException(nameof(propertiesProvider));
+        }
+
+        public string SanitizeName(string name) => name;
+
+        public IReadOnlyCollection<char> GetInvalidCharactersFrom(string name) => Array.Empty<char>();
```

Comment:
Is this right? I see that there's no sanitization or filtering, but then line 32 checks the validity 😕

Reply:
Yes, this is right. EnsureKeyNameIsValid is just asserting string.IsNullOrEmpty. secrets.json allows any character as the config key.